tests/e2e_logging.py

```diff
@@ -774,6 +774,67 @@ def test_historical_query(network, args):
     return network
 
 
+@reqs.description("Attempt to read historical state from missing files")
+@reqs.supports_methods("/app/log/private", "/app/log/private/historical")
+@reqs.no_http2()
+@app.scoped_txs()
+def test_historical_query_on_missing_files(network, args):
+    primary, _ = network.find_primary()
+
+    new_node = network.create_node("local://localhost")
+    network.join_node(
+        new_node,
+        args.package,
+        args,
+        copy_ledger=False,
+        target_node=network.find_any_backup(),
+        from_snapshot=True,
+    )
+    network.trust_node(new_node, args)
+
+    # Pick some sequence numbers before the snapshot the new node started from, and for which
+    # the new node does not have corresponding ledger chunks
+    missing_seqnos = random.sample(
+        range(0, new_node.most_recent_read_only_snapshot_seqno()), 5
```

Comment:
The `GET /node/state` endpoint returns `startup_seqno`, which is the seqno of the snapshot the node started from (`0` if no snapshot). I think this could be used here instead.

Reply:
Nice, I've done this and removed the redundant functions.